---
title: "UMAP"
author: "Max Alcer"
date: "2024-06-06"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(tidyverse)
library(Matrix)
library(umap)
```

### ATAC

```{r}
ATAC_expression_matrix <- readRDS("../data/ATAC-Seq-filtered.rds")
```

```{r}
ATAC_annotations <- read_csv("../data/ATAC_annotations.csv", show_col_types = FALSE)
```

```{r}
custom.config <- umap.defaults
custom.config$verbose <- 1
custom.config$n_neighbors <- 100
umap <- umap(t(ATAC_expression_matrix), 
             config = custom.config)
```


```{r}
ATAC_umap <- tibble("umap_1" = umap$layout[,1],
                    "umap_2" = umap$layout[,2], 
                    "Celltype" = ATAC_annotations$Celltype)
```

```{r}
ggplot(ATAC_umap, aes(x = umap_1, y = umap_2, color = Celltype)) +
  geom_point()
ggsave("plots/UMAP_ATAC.png")
```

### RNA

```{r}
RNA_expression_matrix <- readRDS("../data/pre_processed/RNA-Seq-filtered.rds")
```

```{r}
RNA_annotations <- read_csv("../data/pre_processed/RNA_annotations.csv")
```
```{r}
DEG <- read_csv("../DESeq/genes_by_sig.csv")
```
```{r}
DEG <- DEG %>% 
  filter(padj <= 0.05) %>%
  distinct(gene, keep_all = TRUE) %>%
  head(200)
```

```{r}
RNA_expression_matrix <- RNA_expression_matrix[DEG$gene,]
```

```{r}
custom.config <- umap.defaults
#custom.config$n_neighbors <- 5
#custom.config$min_dist <- 0.5
umap <- umap(t(RNA_expression_matrix), config = custom.config)
```

```{r}
RNA_umap <- tibble(umap_1 = umap$layout[,1], 
                   umap_2 = umap$layout[,2], 
                   Celltype = RNA_annotations$Sub_trajectory_name)
```

```{r}
ggplot(RNA_umap, aes(x = umap_1, y = umap_2, color = Celltype)) +
  geom_point()
ggsave("plots/UMAP_RNA.png")
```

